File: plantio.domain/Generic/DomainException.cs
namespace plantio.Domain.Generic {
    public sealed class DomainException : System.Exception {
        public DomainError Error { get; }

        private DomainException(DomainError error) {
            this.Error = error;
        }

        public static DomainException FromError(DomainError error) {
            return new DomainException(error);
        }
    }
}

File: plantio/src/main/java/com/deltegui/plantio/common/DomainException.java
package com.deltegui.plantio.common;

public class DomainException extends RuntimeException {
	private static final long serialVersionUID = 1L;

	private final DomainError error;

    private DomainException(DomainError error) {
        super(error.getMessage());
        this.error = error;
    }

    public static DomainException fromError(DomainError error) {
        return new DomainException(error);
    }

    public DomainError getError() {
        return error;
    }
}
